WIP: (baby-project-web) Setup

AndreasNasman opened the pull request:
~~Trying to figure out how to add Stylelint to a Vue project. `npx @vue/cli add stylelint` adds most of the dependencies and files needed, but it's missing from any lint checking or formatting.~~ **Skipping stylelint for now since we're using Tailwind CSS.**

Commit:
(baby-project-web) Setup

Commit:
Add files from `vue-ts` template Vite setup

dusan-turajlic commented on src/App.vue:
```diff
@@ -1,30 +1,27 @@
-  color: #42b983;
-}
-</style>
+</style>
```
Missing extra line at the end of file

AndreasNasman replied:
Adding ESLint and Prettier is part of the issue. Once they're added I'll format all files.

dusan-turajlic commented on src/components/HelloWorld.vue:
```diff
@@ -1,65 +1,65 @@
+      required: true
+    }
+  },
+  setup: () => {
```
Shouldent this use the `<script lang="ts" setup="">` syntax insted of this?

AndreasNasman replied:
Yes. The current commit only has files generated straight from Vite with the `vue-ts` template.

dusan-turajlic commented on package.json:
```diff
@@ -1,54 +1,19 @@
-      "vue-cli-service lint",
-      "git add"
-    ]
+    "@vitejs/plugin-vue": "^1.2.1",
```
Hmm with this change we also lost the `mocha+chai` testing setup and the PWA

dusan-turajlic replied:
It is propobly out side the scope of this PR to add thouse back in because we want to have thouse thing here. Right?

dusan-turajlic replied:
Did we also louse the router? We can skip vuex propobly but the router we do need.

AndreasNasman replied:
I'll add Vue Router back. My idea with this issue is to migrate the whole Vue CLI-created project with all the ticketed options to Vite.

dusan-turajlic commented on package.json:
```diff
@@ -1,54 +1,19 @@
   "name": "baby-project-web",
-  "version": "0.1.0",
-  "private": true,
+  "version": "0.0.0",
```
Maybe keept the minor 1 like it had before

AndreasNasman replied:
👍

dusan-turajlic commented on src/App.vue:
```diff
@@ -1,30 +1,27 @@
 
+<script lang="ts">
+import { defineComponent } from 'vue'
+import HelloWorld from './components/HelloWorld.vue'
```
Somthing more to think about is the bring back the resolvers like webpack has. IE the `@` for `/src` but we can make a seperate issue for that

AndreasNasman replied:
Yeah, I'll try and set an @-alias 👍

dusan-turajlic reviewed (commented):
Looks good over all had some conserns and questions that might be out of the scope of this PR but felt like they needed to be raised.

Commit:
Add and initialize ESLint

Commit:
Add Prettier

Commit:
Configure ESLint and Prettier

dusan-turajlic committed:
(baby-project-web) Adds inital issue templates (#3)

Commit:
Add EditorConfig

Commit:
Run Prettier on all applicable files

Commit:
Add @types/node for better type support

Commit:
Fix linting issues

Commit:
Add minor version number

Commit:
Add `@` alias

AndreasNasman commented:
Chosen Vue CLI options for reference
![image](https://user-images.githubusercontent.com/17124245/113931386-12a01a80-97fb-11eb-983c-cbe498e86f05.png)

dusan-turajlic reviewed (approved):
I'm gonna add a Approval for now, but ping me when everything is ready so I can give it one last look see 👍🏼

Commit:
Bump version number in lock file

Commit:
Add Vue Router

Commit:
Add PWA plugin

Commit:
Add ignore paths for formatting and linting

Commit:
Sort imports

Commit:
Fix `@` alias

Commit:
Sort imports and keys

Commit:
Update `index.html` title

Commit:
Refactor to `<setup script>`

https://github.com/vuejs/rfcs/blob/sfc-improvements/active-rfcs/0000-sfc-script-setup.md

Commit:
Add Tailwind CSS

Commit:
Fix ESLint file formats

AndreasNasman committed:
Center image and clean up
